src/Microsoft.DotNet.Interactive.SqlServer/ToolsServiceKernel.cs

```diff
@@ -131,7 +131,7 @@ public async Task HandleAsync(SubmitCode command, KernelInvocationContext contex
             {
                 try
                 {
-                   
+                    var results = new List<TabularDataResource>();
```

Comment:
You're calling StoreQueryResults for each summary still which is at best unnecessary and at worst could result in multiple names with the same list (if it's even possible for the name to change...which seems unlikely but still). 

You could move that call out of the outer foreach right?

Reply:
name collision is left to the implementation. Storing at each bach ensure nothing is lost dueto exceptions or cancellation

Reply:
Why not call store right after creating the list then? It just seems unnecessary to call store each time when you're passing in the same list every time